source/common/orca/orca_load_metrics.cc

```diff
@@ -0,0 +1,80 @@
+#include "source/common/orca/orca_load_metrics.h"
+
+#include <string>
+
+#include "source/common/orca/orca_parser.h"
+
+#include "absl/strings/match.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Orca {
+namespace {
+// The following fields are the names of the metrics tracked in the ORCA load
+// report proto.
+static constexpr absl::string_view kApplicationUtilizationField = "application_utilization";
+static constexpr absl::string_view kCpuUtilizationField = "cpu_utilization";
+static constexpr absl::string_view kMemUtilizationField = "mem_utilization";
+static constexpr absl::string_view kEpsField = "eps";
+static constexpr absl::string_view kRpsFractionalField = "rps_fractional";
+static constexpr absl::string_view kNamedMetricsFieldPrefix = "named_metrics.";
+static constexpr absl::string_view kRequestCostFieldPrefix = "request_cost.";
+static constexpr absl::string_view kUtilizationFieldPrefix = "utilization.";
+} // namespace
+
+void addOrcaNamedMetricToLoadMetricStats(const Protobuf::Map<std::string, double>& metrics_map,
+                                         const absl::string_view metric_name,
+                                         const absl::string_view metric_name_prefix,
+                                         Upstream::LoadMetricStats& stats) {
+  absl::string_view metric_name_without_prefix = absl::StripPrefix(metric_name, metric_name_prefix);
+  // If the metric name is "*", add all metrics from the map.
+  if (metric_name_without_prefix == "*") {
+    for (const auto& [key, value] : metrics_map) {
+      stats.add(absl::StrCat(metric_name_prefix, key), value);
+    }
+  } else {
+    // Add the metric if it exists in the map.
+    const auto metric_it = metrics_map.find(metric_name_without_prefix);
+    if (metric_it != metrics_map.end()) {
+      stats.add(metric_name, metric_it->second);
+    }
+  }
+}
+
+void addOrcaLoadReportToLoadMetricStats(const OptRef<const LrsReportMetricNames>& metric_names,
+                                        const xds::data::orca::v3::OrcaLoadReport& report,
+                                        Upstream::LoadMetricStats& stats) {
+  // If there is no config, then add all `named_metrics`.
+  if (!metric_names.has_value()) {
+    addOrcaNamedMetricToLoadMetricStats(report.named_metrics(), "named_metrics.*",
+                                        kNamedMetricsFieldPrefix, stats);
+    return;
+  }
```

Comment:
A little weird logic.
The whole adding/processing will be skipped if the metrics names from cluster has no values.

And at here, if the metrics names has no values, you will to add all metrics to the stats. 
I think this is a logic confliction this this will never be reached.